LEAF_Request_Portal/Email.php

```diff
@@ -186,24 +184,19 @@ public function sendMail()
         $email['headers'] = html_entity_decode($this->getHeaders(), ENT_QUOTES);
 
         $emailCache = serialize($email);
-        $emailQueueName = sha1($emailCache . random_int(0, 99999999));
         if (strlen(trim($emailCache)) == 0)
         {
             trigger_error('Mail error: ' . $this->emailSubject);
 
             return false;
         }
-        file_put_contents($currDir . '/templates_c/mailer/' . $emailQueueName, $emailCache);
 
-        if (strtoupper(substr(php_uname('s'), 0, 3)) == 'WIN')
-        {
-            $shell = new COM('WScript.Shell');
-            $shell->Run("php {$currDir}/mailer/mailer.php {$emailQueueName}", 0, false);
-        }
-        else
-        {
-            exec("php {$currDir}/mailer/mailer.php {$emailQueueName} > /dev/null &");
-        }
+        require_once dirname(__FILE__) . '/enqueue/EmailProducer.php';
+        require_once dirname(__FILE__) . '/enqueue/EmailConsumer.php';
+        $emailProducer = new EmailProducer();
+        $emailProducer->sendToQueue($emailCache);
+        $emailConsumer = new EmailConsumer();
```

Comment:
What is the effect of having multiple copies of the software be both the EmailProducer and EmailConsumer?

Makes me think EmailConsumer should be a separate service, or in a script that is periodically invoked for exmaple via CronJob